sputnikdao-factory2/src/lib.rs

```diff
@@ -124,22 +150,54 @@ impl SputnikDAOFactory {
     pub fn get_owner(&self) -> AccountId {
         AccountId::new_unchecked(
             String::from_utf8(
-                env::storage_read(OWNER_KEY)
+                env::storage_read(FACTORY_OWNER_KEY)
                     .unwrap_or(env::current_account_id().as_bytes().to_vec()),
             )
             .expect("INTERNAL_FAIL"),
         )
     }
 
-    pub fn get_latest_code_hash(&self) -> Base58CryptoHash {
-        slice_to_hash(&env::storage_read(LATEST_CODE_HASH_KEY).expect("Must have code hash"))
+    pub fn get_default_code_hash(&self) -> Base58CryptoHash {
+        slice_to_hash(&env::storage_read(DEFAULT_CODE_HASH_KEY).expect("Must have code hash"))
     }
 
     /// Returns non serialized code by given code hash.
     pub fn get_code(&self, code_hash: Base58CryptoHash) {
         self.factory_manager.get_code(code_hash);
     }
 
+    pub fn store_contract_metadata(&self, metadata: DaoContractMetadata, set_default: bool) {
+        self.assert_owner();
+        let code_hash: CryptoHash = metadata.code_hash.into();
+        let storage_metadata = env::storage_read(CODE_METADATA_KEY);
+
+        if storage_metadata.is_none() {
+            let storage_metadata = vec![metadata];
+            let serialized_metadata =
+                BorshSerialize::try_to_vec(&storage_metadata).expect("INTERNAL_FAIL");
```

Comment:
This is less important for the factory, but as a reminder, `expect` actually bloats the smart contract a little, as it needs to capture information on line numbers and other info. So it's good to build the reflex of avoiding it for a slightly more strange syntax.

Instead of:

    let owner_id = self.owner_by_id.get(&token_id).expect("Token not found");

it's best to have:

```rust
fn expect_token_found<T>(option: Option<T>) -> T {
  option.unwrap_or_else(|| env::panic_str("Token not found"))
}
let owner_id = expect_token_found(self.owner_by_id.get(&token_id));  
```

As shown in our handy docs site:
https://www.near-sdk.io/reducing-contract-size/examples

Reply:
That information is very handy, but in this case I will chose readability over performance Mike.

That's only because `store_contract_metadata` will be called so rarely (maybe once every half a year?), that the performance increase is insignificant.